components/scream/src/physics/share/physics_universal_impl.hpp

```diff
@@ -46,44 +46,75 @@ Functions<S,D>::get_exner(const Spack& P, const Smask& range_mask)
 }
 //-----------------------------------------------------------------------------------------------//
 // Converts temperature to potential temperature using Exners function:
-//   th_atm = T_atm/exner,
+//   th_mid = T_mid/exner,
 // where
-//   th_atm is the potential temperature, K
-//   T_atm  is the temperature, K
+//   th_mid is the potential temperature, K
+//   T_mid  is the temperature, K
+//   p_mid  is the pressure, Pa  -> used to calculate exners formula.
 //   exner  is the exners formula, see definition above, unitless
 template <typename S, typename D>
 KOKKOS_FUNCTION
 typename Functions<S,D>::Spack
-Functions<S,D>::T_to_th(const Spack& T_atm, const Spack& exner, const Smask& range_mask)
+Functions<S,D>::get_potential_temperature(const Spack& T_mid, const Spack& p_mid, const Smask& range_mask)
 {
   Spack result;
-  const Spack th_atm = T_atm/exner;
+  const Spack exner  = get_exner(p_mid,range_mask);
+  const Spack th_mid = T_mid/exner;
   // Check that there are no obvious errors in the result.
-  check_temperature(th_atm,"T_to_th",range_mask);
+  check_temperature(th_mid,"get_potential_temperature",range_mask);
   // Set the values of the result
-  result.set(range_mask,th_atm);
+  result.set(range_mask,th_mid);
   return result;
 }
+
+template <typename S, typename D>
+KOKKOS_FUNCTION
+template <typename InputProvider>
+void Functions<S,D>::get_potential_temperature(const int nlev,
+                                               const InputProvider& T_mid,
+                                               const InputProvider& p_mid,
+                                               const view_1d<S>& T_potential)
+{
+  Kokkos::parallel_for("get_potential_temperature",nlev,[&](const int& ilev) {
+    T_potential[ilev] = get_potential_temperature(T_mid[ilev],p_mid[ilev],Smask(true))[0];
```

Comment:
I'm pretty sure you don't want the `[0]` at the end. Otherwise, you are stuffing the 1st entry of the result into all entries of the output pack.